### DISCUSSION TITLE
Authentication and authorization for JetStream controller

### DISCUSSION THREAD

**Discussion by sdc395** [nats-io/k8s#429] [Q&A] [answered]
Hi all

I'm attempting to deploy the JetStream Kubernetes controller and running into a problem with authentication/authorization (I can't tell which, specifically).  I'm using a static configuration using NKeys generated using `nk` (I've also tried using `nsc` but with the same result).

The controller logs... `Error: failed to connect to nats: nats: Authorization Violation`, the server logs... 
```
"v1.12.1:go:jetstream-controller" - authentication error - Nkey ""
"v1.12.1:go:jetstream-controller" - ->> [-ERR Authorization Violation]
"v1.12.1:go:jetstream-controller" - Client connection closed: Authentication Failure
```

Could someone please point me in the direction of a beginning-to-end tutorial or minimal working configuration to get the controller working?  I'm really struggling to piece together all the information required (I'm not even certain it's possible using an NKey seed in the credentials file for the controller).

Thanks
Simon

**Comment by wallyqs** [comment, marked as answer]
Not for nkeys, but here is a small guide of setting TLS auth and the jetstream controller: https://gist.github.com/wallyqs/8c4ac3d752edbbf1acfcc8656e8232f2
